algebird-core/src/main/scala/com/twitter/algebird/lsh/vector/VectorMath.scala

```diff
@@ -0,0 +1,87 @@
+package com.twitter.algebird.lsh.vector
+
+/**
+ * Vector math functions for LshVectors.
+ */
+object VectorMath {
```

Comment:
Also note we have VectorSpace: https://github.com/twitter/algebird/blob/develop/algebird-core/src/main/scala/com/twitter/algebird/VectorSpace.scala

Seems you want to add InnerProductSpace:
http://en.wikipedia.org/wiki/Inner_product_space
```scala
trait InnerProductSpace[F, C[_]] extends VectorSpace[F, C] {
  def innerProduct(left: C[F], right: C[F]): F
}
```
Then you can implement such a space for `InnerProductSpace[Double, Array]` or something `InnerProductSpace[Double, Vector]`.